sentry-android-core/src/main/java/io/sentry/android/core/EnvelopeFileObserver.java

```diff
@@ -56,7 +58,11 @@ public void onEvent(int eventType, @Nullable String relativePath) {
     // TODO: Only some event types should be pass through?
 
     final CachedEnvelopeHint hint = new CachedEnvelopeHint(flushTimeoutMillis, logger);
-    envelopeSender.processEnvelopeFile(this.rootPath + File.separator + relativePath, hint);
+
+    final Map<String, Object> hintMap = new HashMap<>();
+    hintMap.put("sentrySdkHint", hint);
```

Comment:
What about this one for those hints we used type check for logic:

```suggestion
    hintMap.put("Sentry:TypeCheckHint", hint);
```

Reply:
That makes sense since it's internal